client/my-sites/site-settings/form-general.jsx

```diff
@@ -170,28 +171,56 @@ class SiteSettingsFormGeneral extends Component {
 		} );
 	};
 
+	renderLanguagePickerNotice = () => {
+		const { fields, translate } = this.props;
+
+		const langId = get( fields, 'lang_id', '' );
+		const matches = /^error_(\w+)$/.exec( langId );
```

Comment:
I'd usually recommend against regular expression checks against like this if we can avoid them. In addition, we could make the method a bit simpler if we refer to the error types specifically and use a little less variables, for example:

```
renderLanguagePickerNotice = () => {
	const { fields, translate } = this.props;
	const langId = get( fields, 'lang_id', '' );
	const errors = {
		error_cap: translate(
			'The Site Language setting is disabled due to insufficient permissions.'
		),
		error_const: translate(
			'The Site Language setting is disabled because your site has the WPLANG constant set.'
		),
	};

	return errors[ langId ] && <Notice text={ errors[ langId ] } isCompact />;
};
```